File: snow_gushi/question2/InterestingParty.py
class InterestingParty():
    def bestInvitation(self, first, second):
        first = list(first)
        second = list(second)
        count_dict = {}
        for f, s in zip(first, second):
            if f not in count_dict:
                count_dict[f] = 1
            else:
                counter = count_dict[f]
                count_dict[f] = counter + 1
            if s not in count_dict:
                count_dict[s] = 1
            else:
                counter = count_dict[s]
                count_dict[s] = counter + 1
        return max(count_dict.values())

File: chooblarin/question02/InterestingParty.java
import java.util.HashMap;
import java.util.Collections;

public class InterestingParty {
  public int bestInvitation(String[] first, String[] second) {
    HashMap<String, Integer> map = new HashMap<String, Integer>();

    for (int i = 0; i < first.length; i++) {
      String fst = first[i];
      int n = (map.get(fst) == null) ? 0 : map.get(fst);
      map.put(fst, n + 1);

      String snd = second[i];
      n = (map.get(snd) == null) ? 0 : map.get(snd);
      map.put(snd, n + 1);
    }
    return Collections.max(map.values());
  }
}
